windows-agent/cmd/ubuntu-pro-agent/agent/agent.go

```diff
@@ -266,3 +276,25 @@ func (a *App) setUpLogger(ctx context.Context) (func(), error) {
 
 	return func() { _ = f.Close() }, nil
 }
+
+func (a *App) ensureSingleInstance(opt options) (cleanup func(), err error) {
+	priv, err := a.privateDir(opt)
+	if err != nil {
+		return nil, fmt.Errorf("could not access the agent's private dir: %v", err)
+	}
+
+	path := filepath.Join(priv, "ubuntu-pro-agent.lock")
+	f, err := createLockFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	pid := strconv.Itoa(os.Getpid())
+	if _, err := f.WriteString(pid); err != nil {
```

Comment:
We are writing the pid inside the lock file, you should really flush it as we are not closing the file right away. Otherwise, we are unsure when the pid will be really written on disk.